src/os_posix/os_union_fs.c

```diff
@@ -1006,21 +1009,32 @@ __union_fs_open_file(WT_FILE_SYSTEM *fs, WT_SESSION *wt_session, const char *nam
             WT_ERR(__union_fs_open_in_source(union_fs, session, union_fh, flags));
 
             if (!dest_exist) {
-                // We're creating a new destination file which is backed by a source file. It
-                // currently has a length of zero, but we want its length to be the same as the
-                // source file. We do this by reading the last byte of the file. This will read
-                // promote the byte into the destination file, setting the file size.
+                /*
+                 * We're creating a new destination file which is backed by a source file. It
+                 * currently has a length of zero, but we want its length to be the same as the
+                 * source file. We do this by reading the last byte of the file. This will read
+                 * promote the byte into the destination file, setting the file size.
+                 */
                 wt_off_t source_size;
-                char buf[10];
 
                 union_fh->source->fh_size(union_fh->source, wt_session, &source_size);
                 __wt_verbose_debug1(session, WT_VERB_FILEOPS,
                   "Creating destination file backed by source file. Copying size (%ld) from source "
                   "file",
                   source_size);
 
-                WT_ERR(__union_fs_file_read(
-                  (WT_FILE_HANDLE *)union_fh, wt_session, source_size - 1, 1, buf));
+                /*
+                 * Set size by truncating. We're bypassing the union layer so we don't track the
+                 * write.
+                 */
+                union_fh->destination.fh->fh_truncate(
```

Comment:
This is a positive length truncate then? Can we elaborate a bit in the comment above so it's more obvious to future readers how this works.

Reply:
Done in [9f8e50e](https://github.com/wiredtiger/wiredtiger/pull/11268/commits/9f8e50e9e9e25c96dd296661234032fd94c5c45c)